modules/objdetect/src/qrcode.cpp

```diff
@@ -70,154 +68,153 @@ void QRDetect::init(const Mat& src, double eps_vertical_, double eps_horizontal_
+                                input_test_lines.push_back(temp_length);
+                                temp_length = 0;
+                                if (input_test_lines.size() == 6) { break; }
+                            }
+                        }
+
+                        // --------------- Compute vertical line --------------- //
+                        if (input_test_lines.size() == 6)
+                        {
+                            double vert_length = 0.0, vert_weight = 0.0;
+
+                            for (size_t coord_i = 0; coord_i < input_test_lines.size(); coord_i++) { vert_length += input_test_lines[coord_i]; }
+
+                            CV_Assert(vert_length > 0);
+                            for (size_t coord_i = 0; coord_i < input_test_lines.size(); coord_i++)
+                            {
+                                if (coord_i % 3 == 0) { vert_weight += fabs((input_test_lines[coord_i] / vert_length) - 3.0/14.0); }
+                                else                  { vert_weight += fabs((input_test_lines[coord_i] / vert_length) - 1.0/ 7.0); }
+                            }
+
+                            if(vert_weight < m_eps_horizontal)
+                            {
+                                local_result.push_back(Point2f(static_cast<float>(line[0] + line[2] * 0.5),
+                                                               static_cast<float>(line[1])));
+                            }
+                        }
+                    }
+                }
             }
         }
 
-        // --------------- Compute vertical lines --------------- //
-
-        if (test_lines.size() == 6)
+        if (local_result.size())
         {
-            double length = 0.0, weight = 0.0;
-
-            for (size_t i = 0; i < test_lines.size(); i++) { length += test_lines[i]; }
-
-            CV_Assert(length > 0);
-            for (size_t i = 0; i < test_lines.size(); i++)
-            {
-                if (i % 3 == 0) { weight += fabs((test_lines[i] / length) - 3.0/14.0); }
-                else            { weight += fabs((test_lines[i] / length) - 1.0/ 7.0); }
-            }
-
-            if(weight < eps_horizontal)
-            {
-                result.push_back(list_lines[pnt]);
-            }
+            m_mtx->lock();
+            m_result.insert(m_result.end(), local_result.begin(), local_result.end());
+            m_mtx->unlock();
         }
     }
-
-    vector<Point2f> point2f_result;
-    for (size_t i = 0; i < result.size(); i++)
-    {
-        point2f_result.push_back(
-              Point2f(static_cast<float>(result[i][0] + result[i][2] * 0.5),
-                      static_cast<float>(result[i][1])));
-    }
-    return point2f_result;
-}
+    ParallelSearchLines& operator=(const ParallelSearchLines &) {
+        return *this;
+    };
+private:
+    Mat m_img;
+    vector<Point2f> &m_result;
+    double m_eps_vertical;
+    double m_eps_horizontal;
+    Mutex *m_mtx;
```

Comment:
It is better to use reference because we don't expect NULL here.
Also use "mutable" specifier.